docs/SPEC.md

```diff
@@ -2793,6 +2915,49 @@ A query specification defines what data and operations to perform.
 The execution model reserves the right to perform those operations as efficiently as possible.
 The execution model may rewrite the query in anyway it sees fit while maintaining correctness.
 
+## Handling data errors
+
+Data errors are errors that occur when processing data from streams.
+When a data error occurs there are a few methods for handling the errors.
+
+First, there is the `fail` function that terminates the program.
+This can be used whenever an error occurs that is not recoverable in any form.
+
+Second, there is the `errors` function to provides direct access to the errors.
+Using the `errors` function any behavior Flux can describe is a possible method for handling errors.
+
+There is an option `errorHandler` that defines the default behavior for all data errors.
+The option value is a transformation function that consumes errors produced by other transformations.
+
+The default behavior is to fail when an error occurs.
+
+    option errorHandler = (tables=<-) =>
+        tables
+            // Fail on any error
+            |> map(fn: (r) => fail(message: r.message))
+
+    // Change the default to log errors to the query system bucket.
+    // Maybe this should just take an explicit bucket instead of having a system bucket?
+    option errorHandler = influxdb.log
+
+The errorHandler transformation is automatically appended to the end of any transformation chain.
+To disable this global behavior set the errorHandler to a noop:
+
+    // Disable global error handling
+    option errorHandler = (tables=<-) => tables
+
+Using the `errors` transformation function specific error handling may be applied to a specific section of a Flux program.
+
+    // Perform a simple downsample
+    from(bucket:"telegraf/autogen")
+        |> range(start: -5m)
+        |> aggregateWindow(every: 1m, fn: mean)
+        |> to(bucket: "telegraf/5m")
+        |> errors()
```

Comment:
once you do this, you commit to losing the data channel.  Any way to recover it?

Reply:
No, but there is a way to preserve it before you lose it. You can assign the result to a variable, then in one branch inspect the errors and in another branch inspect the original data.

```
data = from() |> ...

data 
    |> errors()

data
    |> doSomething() //  this branches errors are not handled by the above branch since they are downstream for that specific error handling.

```